pkg/operator/ceph/cluster/cephstatus.go

```diff
@@ -195,3 +195,31 @@ func (c *ClusterController) updateClusterCephVersion(image string, cephVersion c
 		return
 	}
 }
+
+func (c *ClusterController) UpdateCephStorage(client client.Client, name types.NamespacedName) {
+	cephCluster := &cephv1.CephCluster{}
+	var devices []string
+	err := c.client.Get(context.TODO(), c.namespacedName, cephCluster)
+	if err != nil {
+		if kerrors.IsNotFound(err) {
+			logger.Debug("CephCluster resource not found. Ignoring since object must be deleted.")
+			return
+		}
+		logger.Errorf("failed to retrieve ceph cluster %q to update ceph Storage. %v", c.namespacedName.Name, err)
+		return
+	}
+	cephClusterStorage := cephv1.CephStorage{}
+	devices, err = cephclient.ListDeviceclass(c.context, c.namespacedName.Namespace)
```

Comment:
Let's query the device classes as the first step in this method. We need to reduce the time as much as possible between calling `c.client.Get()` and `UpdateStatus()` to reduce failures when multiple goroutines could be updating the status at the same time.

Reply:
Noted, fixed!